crates/geometry/src/line.rs

```diff
@@ -3,83 +3,70 @@ use std::ops::Mul;
 use path_serde::{PathDeserialize, PathIntrospect, PathSerialize};
 
 use crate::{direction::Direction, line_segment::LineSegment, Distance};
 
 #[derive(
-    Copy,
-    Clone,
-    Debug,
-    Default,
-    Deserialize,
-    Serialize,
-    PathSerialize,
-    PathIntrospect,
-    PathDeserialize,
+    Copy, Clone, Debug, Deserialize, Serialize, PathSerialize, PathIntrospect, PathDeserialize,
 )]
-pub struct Line<Frame, const DIMENSION: usize>(
-    pub Point<Frame, DIMENSION>,
-    pub Point<Frame, DIMENSION>,
-);
+pub struct Line<Frame, const DIMENSION: usize> {
+    pub point: Point<Frame, DIMENSION>,
+    pub direction: Vector<Frame, DIMENSION>,
+}
 
 pub type Line2<Frame> = Line<Frame, 2>;
 pub type Line3<Frame> = Line<Frame, 3>;
 
 impl<Frame> Line2<Frame> {
     pub fn slope(&self) -> f32 {
-        let difference = self.0 - self.1;
-        difference.y() / difference.x()
+        self.direction.y() / self.direction.x()
     }
 
     pub fn y_axis_intercept(&self) -> f32 {
-        self.0.y() - (self.0.x() * self.slope())
-    }
-
-    pub fn project_onto_segment_in_x_axis(&self, point: Point2<Frame>) -> f32 {
-        let rise = (point.x() - self.0.x()) * self.slope();
-        rise + self.0.y()
+        self.point.y() - (self.point.x() * self.slope())
     }
 
     pub fn is_above(&self, point: Point2<Frame>) -> bool {
-        point.y() >= self.project_onto_segment_in_x_axis(point)
+        self.signed_distance_to_point(point) >= 0.0
     }
 
     pub fn signed_distance_to_point(&self, point: Point2<Frame>) -> f32 {
-        let line_vector = self.1 - self.0;
         let normal_vector = Direction::Counterclockwise
-            .rotate_vector_90_degrees(line_vector)
+            .rotate_vector_90_degrees(self.direction)
             .normalize();
-        normal_vector.dot(point.coords()) - normal_vector.dot(self.0.coords())
+        normal_vector.dot(point - self.point)
     }
 
     pub fn intersection(&self, other: &Line2<Frame>) -> Point2<Frame> {
-        let x1 = self.0.x();
-        let y1 = self.0.y();
-        let x2 = self.1.x();
-        let y2 = self.1.y();
-        let x3 = other.0.x();
-        let y3 = other.0.y();
-        let x4 = other.1.x();
-        let y4 = other.1.y();
-
-        point!(
-            ((((x1 * y2) - (y1 * x2)) * (x3 - x4)) - ((x1 - x2) * ((x3 * y4) - (y3 * x4))))
-                / (((x1 - x2) * (y3 - y4)) - ((y1 - y2) * (x3 - x4))),
-            ((((x1 * y2) - (y1 * x2)) * (y3 - y4)) - ((y1 - y2) * ((x3 * y4) - (y3 * x4))))
-                / (((x1 - x2) * (y3 - y4)) - ((y1 - y2) * (x3 - x4)))
-        )
+        let point_difference = self.point - other.point;
+
+        let direction_factor = (point_difference.x() * self.direction.y()
+            - point_difference.y() * self.direction.x())
+            / (self.direction.y() * other.direction.x() - self.direction.x() * other.direction.y());
+
+        other.point + other.direction * direction_factor
```

Comment:
How can I check that this implementation is correct?

Reply:
I derived it yesterday (on paper), I have added test cases covering this. Is this sufficient for you?